CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+the git commit message, using the `-s` flag of the `git commit` command. It will
+add a signoff the looks like following line to your commit message:
+
+```txt
+Signed-off-by: John Smith <john.smith@example.com>
+```
+
+You must use your real name and a reachable email address in the signoff.
+
+Alternately, instead of commits signoff, you can also leave a comment on the PR
+with the following statement:
+
+> I agree to the DCO for all the commits in this PR.
+
+Note that this option still requires that your commits are made under your real
+name and a reachable email address.
+
+## Contributing Code
+
+### Finding Issues
+
+Generally, pull requests that address and fix existing GitHub issues are assigned
+higher priority over those that don't. Use the existing issue labels to help you
+identify relevant and interesting issues.
+
+If you think something ought to be fixed but none of the existing issues
+sufficiently address the problem, feel free to
+[raise a new one](https://github.com/kanisterio/kanister/issues/new/choose).
+
+For new contributors, we encourage you to tackle issues that are labeled as
+`good first issues`.
+
+Regardless of your familiarity with this project, documentation help is always
+appreciated.
+
+Once you found an issue that interests you, post a comment to the issue, asking
+the maintainers to assign it to you.
+
+### Coding Standard
+
+In this project, we adhere to the style and best practices established by the Go
+project and its community.
+
+Specifically, this means:
+
+* adhering to guidelines found in the [Effective Go](https://go.dev/doc/effective_go) document
+* following the common [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+
+The [golangci-lint](https://golangci-lint.run/) tool is used to enforce many
+styling and safety rules.
+
+### Creating A Local Build
+
+See the [BUILD.md](BUILD.md) document for instructions on how to build, test and
+run Kanister locally.
+
+### Commit messages
+
+The basic idea is that we ask all contributors to practice
+[good git commit hygiene](https://www.futurelearn.com/info/blog/telling-stories-with-your-git-history)
+to make reviews and retrospection easy. Use your git commits to provide context
+for the reviewers, and the folks who will be reading the codebase in the months
+and years to come.
+
+Finalized commit messages should look similar to the following format:
+
+```text
+Short one line title
+
+An explanation of the problem, providing context, and why the change is being
+made.
+```
+
+### Submitting Pull Requests
+
+When submitting a pull request, it's important that you communicate your intent,
+by clearly:
+
+1. describing the problem you are trying to solve with links to the relevant
+GitHub issues
+1. describing your solution with links to any design documentation and
+discussion
+1. defining how you test and validate your solution
+1. updating the relevant documentation and examples where appropriate
```

Comment:
should it be numbered list or all 1s are fine.

Reply:
I think this renders correctly once merged. See preview!